src/inference/inference.js

```diff
@@ -124,40 +124,72 @@ function independentTTest (numbers, marginOfError = 0.05, twoTailed = true) {
   return {calc: tCalc, crit: tCrit, isSignificant};
 }
 
-function pearsonR (leftData, rightData) {
+function joinLeftRightData(leftData, rightData) {
   if (leftData.length !== rightData.length) {
     throw 'Data arrays with different sizes';
   }
 
   const data = [];
-  const dataSize = leftData.length;
-  for (let i = 0; i < dataSize; i++) {
+  for (let i = 0; i < leftData.length; i++) {
+    if (typeof leftData[i] !== 'number' || typeof rightData[i] !== 'number') {
+      throw 'Data arrays must contain only numbers';
+    }
+
     data.push({
       right: rightData[i],
       left: leftData[i]
     });
   }
+  return data;
+}
+
+function arraySum (array) {
```

Comment:
You can just import the `sum` function on the `src/basic/basic.js` module. 😄

Reply:
I missed that, thanks